presto-main/src/main/java/com/facebook/presto/execution/ClusterSizeMonitor.java

```diff
@@ -224,6 +254,36 @@ public synchronized ListenableFuture<?> waitForMinimumCoordinators()
         return future;
     }
 
+    public synchronized ListenableFuture<?> waitForMinimumCoordinatorSidecars()
+    {
+        if (currentCoordinatorSidecarCount == 1 || !isCoordinatorSidecarEnabled) {
+            return immediateFuture(null);
+        }
+
+        SettableFuture<?> future = SettableFuture.create();
+        coordinatorSidecarSizeFutures.add(future);
+
+        // if future does not finish in wait period, complete with an exception
+        ScheduledFuture<?> timeoutTask = executor.schedule(
+                () -> {
+                    synchronized (this) {
+                        future.setException(new PrestoException(
+                                GENERIC_INSUFFICIENT_RESOURCES,
```

Comment:
Let's create a new error code for this, resources typically is an expression meant to signify "worker resources"
```suggestion
                                NO_SIDECARS,
```

Reply:
maybe NO_CPP_SIDECARS as the error code to be more specific.